File: fibonacci/fibonacci.py

# ----------method 1

import math

n = 9
golden_ratio = 1.618
fibonacci = (golden_ratio ** n - (1 - golden_ratio) **n) / math.sqrt(5) 

fibonacci = round(fibonacci)

print(fibonacci)

# -----------method 2 collecting values in a list

# list1 = []
# a = 0
# b = 1
# c = 1
# while c < 54:
#   c = a + b
#   a = b
#   b = c
#   list1.append(b)
# print(list1)

File: fibonacci/fibonacci.js

// let n = 3;
// const goldenRatio = 1.618;

// let result = (goldenRatio ** n - (1 - goldenRatio) ** n) / Math.sqrt(5);
// result = Math.ceil(result);

// console.log(result);



function fibonacci(num) {
    if (num == 0) 
    return num
    
        let x = 0, y = 1, z, resultArray = [1];
        
        for (i = 2; i <= num; i++) {
            z = x + y;
            x = y;
            y = z;
            
            resultArray.push(y)
        
        }
        console.log(resultArray);// we are collecting every step's number in an array
        return y;
       
}

console.log(fibonacci(3));